faunadb-java/src/test/java/com/faunadb/client/ClientSpec.java

```diff
@@ -3448,6 +3461,55 @@ public void onComplete() {
     capturedEvents.get();
   }
 
+  @Test
+  public void ParallelsQueriesTest() throws ExecutionException, InterruptedException {
+    Collection<Exception> syncCollection = Collections.synchronizedCollection(new ArrayList<>());
+    List<FaunaClient> clientPool = getClientPool();
+    Random rand = new Random();
+
+    for (int j = 0; j < MAX_PARALLEL_ATTEMPT; j++) {
+      ExecutorService executor = Executors.newFixedThreadPool(IN_PARALLEL_VALUE);
+      for (int i = 0; i < IN_PARALLEL_VALUE; i++) {
+        int randomNum =  rand.nextInt((MAX_PARALLEL_ATTEMPT-1));
+        Runnable worker = new RunnableInParallel(clientPool.get(randomNum), syncCollection);
```

Comment:
Instead of creating new runnable instance let's use the already existing wrapper `CompletableFuture`,
in addition you don't need to pass a reference to `syncCollection`, you can run a task and return the exception as `Optional<Exception>` from that task
```
CompletableFuture<Optional<Exception>> future1  
  = CompletableFuture.supplyAsync(
          () -> {
                 // all your code from the run() method goes here
           }
);
```

You can create a list to store the `CompletableFuture` results:
```
List<CompletableFuture<Optional<Exception>>> results = new ArrayList<>();
```